### DISCUSSION TITLE
Is Visual Basic supported by copilot ?

### DISCUSSION THREAD

**Discussion by ranasrule** [community/community#54618] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Just wanted to ask if  Visual Basic is supported by copilot ?

**Comment by Akash1134** [comment]
Hi @ranasrule ,

Do you mean Visual Studio or Visual Basics? GitHub Copilot is trained on all languages that appear in public repositories. For each language, the quality of suggestions you receive may depend on the volume and diversity of training data for that language. For example, JavaScript is well-represented in public repositories and is one of GitHub Copilot's best supported languages. Languages with less representation in public repositories may produce fewer or less robust suggestions.

GitHub Copilot is available as an extension in Visual Studio Code, Visual Studio, Neovim and the JetBrains suite of IDEs. For more information, see "[Getting started with GitHub Copilot](https://docs.github.com/en/copilot/getting-started-with-github-copilot)."

**Reply by Hupbaby** [reply]
It is impossible to get a straight answer as to whether GitHub CoPilot can be used for Microsoft Visual BASIC development in any combination or flavor of Visual Studio Code or Visual Studio 202x Community, Pro et al.

**Reply by erikdeluca** [reply]
GitHub Copilot is available on RStudio too, the best IDE for R's coders

**Reply by zaxboogie** [reply]
Nobody answered the original question!

**Comment by VBFlyhunter** [comment]
No, the language of Visual Basic is not supported, but several languages in the VB.Net package are.

**Comment by Tarosoft** [comment]
They do not support vb.net yet, I try already. Also, no information mentioned to vb.net so far.

**Reply by Tarosoft** [reply, marked as answer]
Hey, guy. Now Github Copilot support Visual Basic. Make sure you have the last update for VS2022. After install the Github Copilot, they understand Visual Basic without any settings. Have fun.

**Reply by tamtranQH** [reply]
Thank you!

**Reply by RobertRobertRobert** [reply]
install codeium instead for now.

**Comment by ghost** [comment]
Visual basic needs to be supported fully as well and github copilot needs to be well trained. I am writing a project in purely visual basic because that's the only programming language I am comfortable with. I spend several hours going around and around with copilot over code with no luck.
